WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java

```diff
@@ -0,0 +1,590 @@
+    public void onPause() {
+        if (isFinishing() && mRetainedFragment != null) {
+            getFragmentManager().beginTransaction().remove(mRetainedFragment).commit();
+        }
+        super.onPause();
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(final MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            setResult(RESULT_CANCELED);
+            finish();
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public boolean onQueryTextSubmit(String query) {
+        if (mSearchView != null) {
+            mSearchView.clearFocus();
+        }
+        ActivityUtils.hideKeyboard(this);
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String query) {
+        if (!StringUtils.equals(query, mSearchQuery)) {
+            submitSearch(query, true);
+        }
+        return true;
+    }
+
+    private void configureSearchView() {
+        mSearchView = findViewById(R.id.search_view);
+
+        // don't allow the SearchView to be closed
+        mSearchView.setOnCloseListener(new SearchView.OnCloseListener() {
+            @Override public boolean onClose() {
+                return true;
+            }
+        });
+
+        mSearchView.setOnQueryTextListener(this);
+    }
+
+    private void showEmptyView(boolean show) {
+        if (!isFinishing()) {
+            TextView txtEmpty = findViewById(R.id.text_empty);
+            txtEmpty.setVisibility(show ? View.VISIBLE : View.GONE);
+            if (show) {
+                boolean isEmpty = mSearchQuery == null || mSearchQuery.length() < MIN_SEARCH_QUERY_SIZE;
+                if (isEmpty) {
+                    String message = getString(R.string.stock_media_picker_initial_empty_text);
+                    String subMessage = getString(R.string.stock_media_picker_initial_empty_subtext);
+                    String link = "<a href='https://pexels.com/'>Pexels</a>";
+                    String html = message
+                                  + "<br /><br />"
+                                  + "<small>" + String.format(subMessage, link) + "</small>";
+                    txtEmpty.setMovementMethod(WPLinkMovementMethod.getInstance());
+                    txtEmpty.setText(Html.fromHtml(html));
+                } else {
+                    txtEmpty.setText(R.string.stock_media_picker_empty_results);
+                }
+            }
+        }
+    }
+
+    private void showProgress(boolean show) {
+        if (!isFinishing()) {
+            findViewById(R.id.progress).setVisibility(show ? View.VISIBLE : View.GONE);
+        }
+    }
+
+    private void submitSearch(@Nullable final String query, boolean delayed) {
+        mSearchQuery = query;
+
+        if (query == null || query.length() < MIN_SEARCH_QUERY_SIZE) {
+            mAdapter.clear();
+            showEmptyView(true);
+            return;
+        }
+
+        if (delayed) {
```

Comment:
We really need to write a util class/method for this. We have the same code (different times) all around our codebase.

Of course it's outside of the scope of this PR :)